### ISSUE TITLE
[V3] Implement new Concept Exercise: classes-intro

### ISSUE THREAD

**Issue by rishiosaur** [exercism/javascript#942]
This issue describes how to implement the `class` concept exercise for the JavaScript track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

This exercise's goal is to teach the student about the function of `class`es in Javascript, why they exist, and how to implement them.

## Learning objectives

- Understand the `constructor` function
- Understand how `this` works
- Know how to effectively create prototypal properties

## Out of scope
- The `new` keyword
- Initializing classes (this exercise will only teach students how to implement classes in this exercise, not how to use them).
- Factory functions
- Class expressions (declaring a class with a `var` statement)
- Static properties & static methods
- `extends` keyword
- Class inheritance with `super`

## Concepts

- Creating private properties in the `constructor` function
- Creating public functions that consume the properties defined in the constructor
- Using `this` to access class-wide properties

## Prerequisites

- `booleans`
- `strings`
- `functions`
- `objects`

### Hints

- [How to use `this` to access class properties][http://test.com]
- [Using `this` in methods on the class prototype](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#Prototype_methods)

### After

- [`Function.prototype.bind`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind)
- [Factory Functions](https://medium.com/javascript-scene/javascript-factory-functions-with-es6-4d224591a8b1)
- [Static Methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#Static_methods)

## Representer

No changes required

## Analyzer

No changes required

## Implementing

See the [implementation guide](https://github.com/exercism/v3/blob/master/languages/javascript/reference/implementing-a-concept-exercise.md#how-to-implement-a-rust-concept-exercise) for JavaScript for more detail.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by junedev** [comment]
A replacement issue for this one was created to fit the target learning curve: https://github.com/exercism/javascript/issues/1015 Please go there if you still want to contribute to the classes concept exercise.
This issue will be closed.
